### DISCUSSION TITLE
Should client object be reused?

### DISCUSSION THREAD

**Discussion by cs-ranbi** [aws/aws-sdk-go-v2#2510] [Q&A] [answered]
Noobs question: for client objects like s3 client, kms client, should I create new client object on demand or should I cache one client object and reuse it whenever possible?

**Comment by iann0036** [comment, marked as answer]
It's optimal to create one client object (per service) and reuse it wherever possible.

In platforms like Lambda, initialisation is recommended to be done [outside of the handler function](https://docs.aws.amazon.com/lambda/latest/operatorguide/static-initialization.html) to slightly optimise the code for future invocations. If you're using a service object pattern within Go, the client is typically stored within the service and used by any associated methods that need it.
